### PR TITLE
fix: warn if code coverage is requested but no tests ran

### PR DESCRIPTION
**Pull request by cristiand391**
### What does this PR do?
Make deploy commands show a warning if code coverage reports are requested but no tests ran during the deployment.

#### Before

User specifies `--coverage-formatters` and/or`--junit` to get codecov and junit tests results but misses `--test-level` so no tests run and reports aren't generated. 

#### After
`sf` will show a msg about setting the test-level for the deploy when `--test-level === undefined` or `--test-level === NoTestRun`.
Screenshot 2023-07-10 at 12 19 04

Screenshot 2023-07-10 at 12 17 57

### What issues does this PR fix or reference?
@W-13725604@

### CODE REVIEW

File: src/formatters/deployResultFormatter.ts [hunk 1]
```diff
@@ -107,17 +108,35 @@ export class DeployResultFormatter extends TestResultsFormatter implements Forma
...
+        testsWarn += `\`--coverage-formatters\` was specified but no tests ran.${EOL}`;
       }
       if (this.junit) {
-        ux.log(`Junit results written to ${this.resultsDir}/junit/junit.xml`);
-        this.createJunitResults();
+        testsWarn += `\`--junit\` was specified but no tests ran.${EOL}`;
       }
+
+      testsWarn +=
+        'You can ensure tests run by specifying `--test-level` to `RunSpecifiedTests`, `RunLocalTests` or `RunAllTestsInOrg`.';
```
Comment (jshackell-sfdc) [inline src/formatters/deployResultFormatter.ts:125]:
```suggestion
        'You can ensure tests run by specifying `--test-level` and setting it to `RunSpecifiedTests`, `RunLocalTests` or `RunAllTestsInOrg`.';
```

Comment (cristiand391) [reply src/formatters/deployResultFormatter.ts]: thanks Juliet 😄

**Review by WillieRuemmele — COMMENTED** [review]
could you add a NUT or UT?

could this be handled by flag validation?

File: src/commands/project/convert/mdapi.ts [hunk 2]
```diff
@@ -132,14 +132,15 @@ export class Mdapi extends SfCommand<ConvertMdapiJson> {
     }
   }
 
-  protected formatResult(): ConvertMdapiJson {
+  // eslint-disable-next-line @typescript-eslint/require-await
```
Comment (WillieRuemmele) [inline src/commands/project/convert/mdapi.ts:135]: is this rule still required?

**Review by WillieRuemmele — APPROVED** [review]

File: src/commands/project/convert/mdapi.ts (hunk 2 above)
Comment (cristiand391) [reply src/commands/project/convert/mdapi.ts]: nope, good catch. removed 👍🏼

**Comment by WillieRuemmele** [comment]
### QA Notes
---
ran command to generate test coverage, but didn't specify any tests, confirmed no warning
✅ : on this branch 
```bash
 ../../oss/plugin-deploy-retrieve/bin/dev  project:deploy:start --source-dir force-app/main/default/classes --coverage-formatters html 
Deploying v58.0 metadata to test-occfrql1lpqk@example.com using the v58.0 SOAP API.
Deploy ID: 0Af7X000011k0oCSAQ
Status: Succeeded | ████████████████████████████████████████ | 9/9 Components (Errors:0) | 0/0 Tests (Errors:0)

Deployed Source
===================================================================================================================
| State     Name                     Type      Path                                                                 
| ───────── ──────────────────────── ───────── ──────────────────────────────────────────────────────────────────── 
...  
| Unchanged TestSampleDataController ApexClass force-app/main/default/classes/TestSampleDataController.cls-meta.xml 

Code Coverage formats, [html], written to coverage/
Warning: `--coverage-formatters` was specified but no tests ran.
You can ensure tests run by specifying `--test-level` and setting it to `RunSpecifiedTests`, `RunLocalTests` or `RunAllTestsInOrg`.
```

and specifying a test level, no warning
✅ :
```bash
 ➜  ../../oss/plugin-deploy-retrieve/bin/dev  project:deploy:start --source-dir force-app/main/default/classes --coverage-formatters html --test-level RunAllTestsInOrg
Deploying v58.0 metadata to test-occfrql1lpqk@example.com using the v58.0 SOAP API.
Deploy ID: 0Af7X000011k0pZSAQ
Status: Succeeded | ████████████████████████████████████████ | 9/9 Components (Errors:0) | 11/11 Tests (Errors:0)

Deployed Source
===================================================================================================================
| State     Name                     Type      Path                                                                 
| ───────── ──────────────────────── ───────── ──────────────────────────────────────────────────────────────────── 
...   
| Unchanged TestSampleDataController ApexClass force-app/main/default/classes/TestSampleDataController.cls-meta.xml 

Test Results Summary
Passing: 11
Failing: 0
Total: 11
Time: 4132
Code Coverage formats, [html], written to coverage/
```

**Pull request merged**